### PR TITLE
Update cabal metadata and CIs

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: .packcheck.ignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.packcheck.ignore
+.github/workflows/haskell.yml
+.gitignore
+.hlint.yaml
+appveyor.yml
+cabal.project.Werror
+stack.yaml
```
Comment (adithyaov) [inline .packcheck.ignore:7]: We generally add these to the dist?
I remember asking this question before.

Comment (adithyaov) [reply .packcheck.ignore:7]: Should we make it a habit to not add the CI files and hlint and stack files to the dist?

File: Benchmark/System/Process.hs [hunk 2]
```diff
@@ -101,7 +91,7 @@ generateByteFile =
                 ]
 
         (_, _, _, procHandle) <- createProcess procObj
-        waitForProcess procHandle
+        _ <- waitForProcess procHandle
```
Comment (adithyaov) [inline Benchmark/System/Process.hs:94]:
```suggestion
        void $ waitForProcess procHandle
```

File: streamly-process.cabal [hunk 3]
```diff
@@ -1,96 +1,103 @@
...
-    stack.yaml
-    CHANGELOG.md
-    README.md
-    NOTICE
-    cabal.project.ci
-    design/proposal.md
+  CHANGELOG.md
+  NOTICE
+  README.md
+  design/proposal.md
```
Comment (adithyaov) [inline streamly-process.cabal:22]: This is cleaner. We should make it a habit to add only the necessary files to the dist.

File: streamly-process.cabal [hunk 4]
```diff
@@ -1,96 +1,103 @@
...
+    -Wredundant-constraints
+    -Wnoncanonical-monad-instances
   build-depends:
-      streamly          >= 0.7.1 && < 0.8
-    , primitive         >= 0.5.4 && < 0.8
-    , base              >= 4.8   && < 5
+      base              >= 4.8   && < 5
     , process           >= 1.0   && < 1.7
-    , transformers      >= 0.4 && < 0.6
+    , streamly          >= 0.7.1 && < 0.8
```
Comment (adithyaov) [inline streamly-process.cabal:63]: Can you open an issue to update streamly and assign it to me? Or are you doing it?

**Review by adithyaov — APPROVED** [review]

File: streamly-process.cabal (hunk 4 above)
Comment (harendra-kumar) [reply streamly-process.cabal:63]: I have updated it in the next change.

File: .packcheck.ignore (hunk 1 above)
Comment (harendra-kumar) [reply .packcheck.ignore:7]: A distribution is everything that is required to compile and use the software. If there were a concept of "dev" distribution, we could package all the development related artifacts in that. But the git repo is available for that purpose.

**Pull request merged**